membership/views.py

```diff
@@ -1179,32 +1178,71 @@ def member_reg_form(request, title, pk):
             # user is not allowed to edit this member
             return redirect('dashboard')
 
+
     elif request.method == "POST":
         # admin/owner can add any new/existing user
         # admin/owner can edit any existing owner
         # user can edit only themselves
-
         form = MemberForm(request.POST)
         if form.is_valid():
-            user = get_or_create_user(form)
             if pk == 0 and request.user == membership_package.owner or request.user in membership_package.admins.all():
+                userExists = User.objects.filter(email=form.cleaned_data['email']).exists()
+                if userExists == False:
+                    User.objects.create(first_name=form.cleaned_data['first_name'],
+                                        last_name=form.cleaned_data['last_name'],
+                                        email=form.cleaned_data['email'],
+                                        username=generate_username(form.cleaned_data['first_name'],
+                                                                   form.cleaned_data['last_name']))
                 # new member
                 # validate user not already a member of package
                 try:
-                    if MembershipSubscription.objects.filter(member=Member.objects.get(user_account=User.objects.get(email=form.cleaned_data['email'])),
-                                                             membership_package=membership_package).exists():
-                        form.add_error('email', f"This email address is already in use for {membership_package.organisation_name}.")
+                    if MembershipSubscription.objects.filter(member=Member.objects.get(
+                            user_account=User.objects.get(email=form.cleaned_data['email'])),
+                            membership_package=membership_package).exists():
+                        form.add_error('email', f"This email address is already in use for "
+                                                f"{membership_package.organisation_name}.")
                 except Member.DoesNotExist:
                     pass
                 except User.DoesNotExist:
                     pass
+
+                user = User.objects.get(email=form.cleaned_data['email'])
```

Comment:
this should be moved up to an else: statement if the user exists. with the current logic it's doing two calls to the database when it only needs to be one.